src/components/ebay-calendar/component.js

```diff
@@ -0,0 +1,383 @@
+        this.state.tabindexISO = iso;
+        // After UI updates, focus on the new tabindex date
+        setTimeout(() => {
+            /** @type {HTMLButtonElement | undefined} */ (this.getEl(iso))?.focus();
+        });
+    }
+
+    /**
+     * @param {Input} input
+     */
+    calculateRangeDisplay(input = this.input) {
+        if (input.selected && input.range) {
+            // Determine range display (state.rangeStart-state.rangeEnd)
+            let iso1, iso2;
+            if (Array.isArray(input.selected)) {
+                [iso1, iso2] = input.selected;
+            } else if (this.state.focusISO) {
+                iso1 = input.selected;
+                iso2 = this.state.focusISO;
+            }
+            if (iso1 && iso2) {
+                if (iso1 < iso2) {
+                    this.state.rangeStart = iso1;
+                    this.state.rangeEnd = iso2;
+                } else {
+                    this.state.rangeStart = iso2;
+                    this.state.rangeEnd = iso1;
+                }
+            } else {
+                this.state.rangeStart = this.state.rangeEnd = undefined;
+            }
+        }
+    }
+}
+
+/**
+ * @param {string} localeName
+ * @return {number} 0 or 7 is Sun, 1 is Mon, -1 or 6 is Sat
+ */
+export function findFirstDayOfWeek(localeName) {
+    // weekInfo only exists on some browsers, so we default to Sunday otherwise
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/weekInfo
+
+    const locale = /** @type {Intl.Locale & { weekInfo?: { firstDay: number } }} */ (
+        new Intl.Locale(localeName)
+    );
+    if (locale.weekInfo) {
+        return locale.weekInfo.firstDay;
+    }
+    return 0;
+}
+
+/**
+ * @param {string} localeName
+ */
+export function getWeekdayInfo(localeName) {
+    const firstDayOfWeek = findFirstDayOfWeek(localeName);
+
+    const weekdayLabelFormatter = new Intl.DateTimeFormat(localeName, {
+        weekday: 'short',
+    });
+    const weekday = new Date(2022, 9, 2 + firstDayOfWeek); // October 2, 2022 was a Sunday
+    const weekdayLabels = [...Array(7)].map(() => {
+        const dayLabel = weekdayLabelFormatter.format(weekday);
+        weekday.setDate(weekday.getDate() + 1);
+        return dayLabel;
+    });
+
+    return { firstDayOfWeek, weekdayLabels };
+}
+
+/**
+ * @param {DateConstructor["arguments"]} arg
+ */
+function dateArgToISO(arg) {
+    return arg ? toISO(new Date(arg)) : undefined;
+}
+
+/**
+ * @param {Date} date
+ * @returns {DayISO}
+ */
+export function toISO(date) {
+    return /** @type {DayISO} */ (date.toISOString().slice(0, 10));
+}
+
+/**
+ * @param {DayISO} iso
+ */
+export function fromISO(iso) {
+    const [year, month, day] = iso.split('-');
+    return new Date(+year, +month - 1, +day);
+}
+
+/**
+ * @param {string | undefined} locale
+ */
+export function localeOverride(locale) {
+    if (locale) return locale;
+
+    return navigator.language;
+}
```

Comment:
I'm tempted to have these in a helper.js file, which we can import from component and in the template. Just for organization.

Reply:
Ok, I ended up just moving it into `helper.js` in `calendar`